### PR TITLE
[ONNX] Refactor validation op-level

### PR DESCRIPTION
**Pull request by titaiwangms**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94920

### CODE REVIEW

File: torch/onnx/_internal/fx/exporter.py [hunk 1]
```diff
@@ -1058,92 +1059,64 @@ def save_model_with_external_data(
...
+    )
     try:
         with evaluator.default_as(evaluator.ort_evaluator):
             expected_outputs = node.target(*torch_args, **torch_kwargs)  # type: ignore[operator]
             # TODO(titaiwang): Expose _convert_tensor_to_numpy and _convert_kwargs_for_onnx?
-            input_onnx = [_convert_tensor_to_numpy(x) for x in torch_args]
+            input_onnx = [
+                onnx_proto_utils._convert_tensor_to_numpy(x) for x in torch_args
+            ]
             # deal with dtype and device
```
Comment (BowenBao) [inline torch/onnx/_internal/fx/exporter.py:1086]: Is this outdated? I think below handles more than these two.

File: torch/onnx/_internal/fx/exporter.py [hunk 2]
```diff
@@ -1058,92 +1059,64 @@ def save_model_with_external_data(
...
+                        if isinstance(torch_output, torch.Tensor)
+                        else torch.tensor(torch_output).cpu(),
+                        rtol=1e-4,
+                        atol=1e-3,
                     )
+
                 except AssertionError as e:
                     warnings.warn(
-                        f"Suppressed AssertionError:\n{e}.\n"
+                        f"\nSuppressed AssertionError:\n{e}.\n"
```
Comment (BowenBao) [inline torch/onnx/_internal/fx/exporter.py:1117]: nit: why start with a newline?

File: torch/onnx/_internal/onnx_proto_utils.py [hunk 3]
```diff
@@ -287,3 +291,25 @@ def _find_onnxscript_op(
                 else None,
             )
     return onnx_function_list, included_node_func
+
+
+def _convert_tensor_to_numpy(input: Any) -> Any:
+    if isinstance(input, torch.Tensor):
```
Comment (BowenBao) [inline torch/onnx/_internal/onnx_proto_utils.py:297]: Moving numpy import to here will resolve CI issue. The rule is not to make `import torch` depend on `numpy`.

File: torch/onnx/_internal/fx/exporter.py [hunk 4]
```diff
@@ -1058,92 +1059,64 @@ def save_model_with_external_data(
...
     node: torch.fx.Node,
-    symbolic_fn: onnxscript.OnnxFunction,
+    symbolic_fn: Union[onnxscript.OnnxFunction, Callable],
     torch_args: tuple,
     torch_kwargs: dict,
 ):
     """Validate the op between ONNX Runtime and PyTorch."""
     # op-level validation
     # Symbolic_fn should have the same output as node.target (torch ops)
+    # TODO: torch.dtype to onnx.dtype conversion
```
Comment (BowenBao) [inline torch/onnx/_internal/fx/exporter.py:1072]: nit: is this done after this PR?

File: torch/onnx/_internal/onnx_proto_utils.py (hunk 3 above)
Comment (titaiwangms) [reply torch/onnx/_internal/onnx_proto_utils.py:297]: Thanks!

File: torch/onnx/_internal/fx/exporter.py [hunk 5]
```diff
@@ -261,7 +262,7 @@ def _wrap_fx_args_as_onnxscript_args(
         _retrieve_or_adapt_input_to_graph_set(arg, fx_name_to_onnxscipt_value)
         for arg in complete_args
     )
-    graph_kwargs = _filter_incompatible_kwargs(complete_kwargs)
+    graph_kwargs = _filter_incompatible_and_dtype_convert_kwargs(complete_kwargs)
```
Comment (BowenBao) [inline torch/onnx/_internal/fx/exporter.py:265]: nit: should we call this `onnxscript_kwargs`?

**Review by BowenBao — APPROVED** [review]
LGTM w/ inline comments.

File: torch/onnx/_internal/fx/exporter.py (hunk 1 above)
Comment (titaiwangms) [reply torch/onnx/_internal/fx/exporter.py:1086]: Yes, I changed the function name to `_filter_incompatible_and_dtype_convert_kwargs` which should cover 100% what we want in this particular comment.

File: torch/onnx/_internal/fx/exporter.py (hunk 2 above)
Comment (titaiwangms) [reply torch/onnx/_internal/fx/exporter.py:1117]: I wanted to separate the warning from each op better. It was super crowded, and I couldn't tell the message was from which op. Ideally, there shouldn't be too many ops popping out warning messages.

**Comment by BowenBao** [comment]
On another note, it would be great if `verification.py` can be re-used if possible.

**Pull request closed without merging**